### ISSUE TITLE
Call to device timed out: Xiaomi Air Purifier 2S (mc1)

### ISSUE THREAD

**Issue by degofedal** [YinHangCode/homebridge-mi-airpurifier#32]
Hi

Thanks for a great plugin. Just got my Xiaomi Air Purifier 2S in the door and succeed in getting the token and plugin up and running with.

The device itself however is not responding in Apple Home app. Quite possibly due to lack of support for model "zhimi.airpurifier.mc1" in miio? [Link](https://github.com/aholstenson/miio/pull/229)

Any chance support for zhimi.airpurifier.mc1 will be added directly to the plugin?

I'm getting a"call to device timed out" in Homebridge:
![Screenshot 2019-04-09 at 21 28 09](https://user-images.githubusercontent.com/3520690/55829784-c9d00100-5b0f-11e9-873c-1ad0b09819d3.png)

Thanks again :)

**Comment by degofedal** [comment]
Just an update on this issue. Fixed it by obtaining a token via [this method](https://github.com/jghaanstra/com.xiaomi-miio/blob/master/docs/obtain_token.md)

Everything now work as it should.

**Comment by bucikowyyy** [comment]
it's work ? becouse I have the same issue. I'm connected straight to 2S and i useing mioo dicover --sync after that i copied token and write in config.json in homebridge. I have fresh install homebridge and i used @lastversion during install miio, packages and homebridge

**Comment by degofedal** [comment]
> it's work ? becouse I have the same issue. I'm connected straight to 2S and i useing mioo dicover --sync after that i copied token and write in config.json in homebridge. I have fresh install homebridge and i used @lastversion during install miio, packages and homebridge

Try getting your token [using this method](https://github.com/jghaanstra/com.xiaomi-miio/blob/master/docs/obtain_token.md). Got my token initially from miio discover but it seems like it is unreliable. Good luck.

**Comment by bucikowyyy** [comment]
Yes you have right working s2 all the time change token when have new connect to wifi. It’s work thank you ;)

**Comment by MickL** [comment]
Thanks i successfully used the iOS method to get the token! I am pretty sure they read and follow this and other repositories. I wouldnt be suprised if they implement better encryption on the token in the iOS sql database. Maybe the best is to not never connect the device to Mi App and never update it.

@degofedal Please close this issue
